packages/jet-brains-integration/src/web-types-generator.ts

```diff
@@ -33,20 +33,33 @@ import { updateConfig } from "../../../tools/configurations";
 const packageJson = JSON.parse(fs.readFileSync("package.json", "utf8"));
 
 export function getTagList(
-  components: Component[],
-  options: Options,
-  referenceTemplate?: (name: string, tag?: string) => Reference
+  components: ComponentWithModule[],
+  options: Options
 ): WebTypeElement[] {
-  return components.map((component: Component) => {
-    const reference = referenceTemplate
-      ? referenceTemplate(component.name, component.tagName)
+  return components.map((component: ComponentWithModule) => {
+    const reference = options.referenceTemplate
+      ? options.referenceTemplate(component.name, component.tagName)
       : undefined;
 
+    const sourceModule = options.sourceModuleTemplate
+      ? options.sourceModuleTemplate({
+          name: component.name,
+          tag: component.tagName,
+          modulePath: component.module.path,
+        })
+      : component.module.path;
+
     return {
       name: `${options.prefix}${
         component.tagName || toKebabCase(component.name)
       }${options.suffix}`,
       description: getComponentDetailsTemplate(component, options),
+      source: sourceModule
+        ? {
+            symbol: component.name,
```

Comment:
If you export your component as default like

`export default class XXXAutocompleteInput extends LitElement {...`

The symbol you have to set is "default" otherwise IntelliJ will not jump to the source file.

Reply:
I'm not sure if there is an indicator on the CEM to identify if a component has a named export or not.

Reply:
I've experimented a bit more, oddly enough it doesn't seem to work on `demo/vanilla-demo`, if I change the extension from .js to .ts and add `export` to the class declaration it works (export alone doesn't work with .js).
Not sure why IntelliJ treats the js files different to the ts